azurerm/resource_arm_dns_a_record.go

```diff
@@ -121,7 +121,7 @@ func resourceArmDnsARecordRead(d *schema.ResourceData, meta interface{}) error {
 
 	resGroup := id.ResourceGroup
 	name := id.Path["A"]
-	zoneName := id.Path["dnszones"]
+	zoneName := id.Path["dnsZones"]
```

Comment:
Can we define a constant string to hold this value instead of updating every occurrence in the near future ?

Reply:
whilst that'd potentially save a few lines here and there, I'd rather we added support in the SDK for parsing Resource ID's to be honest: 
https://github.com/Azure/azure-sdk-for-go/issues/3080#issuecomment-432080173 (which would mean we could remove most of this logic altogether)